### PR TITLE
Unit test to download S1-A/B SAFE file

### PR DESCRIPTION
**Pull request by vbrancat**
This PR adds a unit test to download S1-A/B data. To avoid using credential info, data are downloaded from the AWS s3 seasia bucket containing S1-A/B data covering only South East Asia.

The PR should partially address issue #10. Another PR will follow for a test downloading the XML file and test  various reader functionalities.

Note: To download approx 4 GB of data takes around 10-15 min on average. It is recommended that this test runs nightly as a part of the CI suite.

### CODE REVIEW

File: tests/download_data.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import shutil
+from urllib.request import urlopen
+
+s3_url = "http://sentinel1-slc-seasia-pds.s3-website-ap-southeast-1.amazonaws.com/datasets/slc/v1.1"
+data_info = "2021/08/15"
+granule_id = "S1A_IW_SLC__1SDV_20210815T100025_20210815T100055_039238_04A1E8_D145"
+
+
+def download_granule(in_url):
+    zip_file = in_url.split('/')[-1]
```
Comment (rtburns-jpl) [inline tests/download_data.py:11]:
```suggestion
    zip_file = os.path.basename(in_url)
```

File: tests/download_data.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    if os.path.isfile(zip_file) == False:
```
Comment (rtburns-jpl) [inline tests/download_data.py:12]:
```suggestion
    if not os.path.isfile(zip_file):
```

**Review by rtburns-jpl — APPROVED** [review]
Nice, started a download locally and it seems to be working. The download is pretty big and doesn't show any sign that it is making progress, do you know if there's an easy way to show a progress bar?

**Comment by yunjunz** [comment]
`wget` will print out a nice progress bar as below, although we will have to use the `os.system(wget)` syntax for this.

```
wget https://zenodo.org/record/5498198/files/FernandinaSenDT128.tar.xz
--2022-02-06 04:09:35--  https://zenodo.org/record/5498198/files/FernandinaSenDT128.tar.xz
Resolving zenodo.org (zenodo.org)... 137.138.76.77
Connecting to zenodo.org (zenodo.org)|137.138.76.77|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 277140032 (264M) [application/octet-stream]
Saving to: ‘FernandinaSenDT128.tar.xz’

FernandinaSenDT128. 100%[===================>] 264.30M  10.4MB/s    in 34s     

2022-02-06 04:10:10 (7.79 MB/s) - ‘FernandinaSenDT128.tar.xz’ saved [277140032/277140032]
```

**Comment by LiangJYu** [comment]
SO suggestions:
[`shutil` wrapper](https://stackoverflow.com/a/29967714)
[add/use `requests`](https://stackoverflow.com/a/15645088)

I'm partial to the `shutil` wrapper to keep dependencies light.

**Comment by rtburns-jpl** [comment]
Some distros don't come with wget, but it's a pretty small dependency to add. Sounds like a good approach to me 👍

**Comment by vbrancat** [comment]
@rtburns-jpl before adding a `wget` dependency I can try to see if there is a workaround with `requests`

**Comment by vbrancat** [comment]
Another option (lighter in terms of code to implement) is to have as a dependency `python-wget`

Using:
wget.downloat(in_url)

gives this nice progress bar
Screen Shot 2022-02-07 at 2 33 24 PM

@yunjunz @rtburns-jpl @LiangJYu how does that sound?

**Comment by yunjunz** [comment]
I am not sure, the `python-wget` has not been updated since 2015 and it seems to be abanded by the developer, thus, I am more inclined to use `wget` directly. 

Note that we should use a different dependency list for testing, which is probably longer than the code dependency that end-users need, so I am not very concerned about adding `wget`.

**Comment by vbrancat** [comment]
> I am not sure, the `python-wget` has not been updated since 2015 and it seems to be abanded by the developer, thus, I am more inclined to use `wget` directly.
> 
> Note that we should use a different dependency list for testing, which is probably longer than the code dependency that end-users need, so I am not very concerned about adding `wget`.

Sounds good :)

**Comment by LiangJYu** [comment]
@yunjunz's solution `wget` solution sounds good to me.

Since you've already written this [function](https://github.com/vbrancat/sentinel1-reader/blob/794d092a2e05f2492b223559c3c7b60903b2eff7/tests/download_data.py#L10), why not put this in a function `src/sentinel1_reader`?

**Comment by vbrancat** [comment]
> @yunjunz's solution `wget` solution sounds good to me.
> 
> Since you've already written this [function](https://github.com/vbrancat/sentinel1-reader/blob/794d092a2e05f2492b223559c3c7b60903b2eff7/tests/download_data.py#L10), why not put this in a function `src/sentinel1_reader`? This would address #3.

I have a script (that I have never pushed) to download orbit ephemerides without credentials. Will address this in another PR.

**Review by yunjunz — APPROVED** [review]
Looks great!

**Review by hfattahi — APPROVED** [review]
LGTM!

**Pull request merged**